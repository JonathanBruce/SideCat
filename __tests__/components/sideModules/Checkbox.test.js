/* global test, expect, mount */
// Checkbox.test.js

import React from 'react';
import Checkbox from 'js/components/sideModules/CheckBox';
import renderer from 'react-test-renderer';

test('Checkbox changes the class when it is checked by default', () => {
	const component = renderer.create(
		<Checkbox label='foo' checked={ true } />
	);

	let tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});

test('Checkbox\'s parent component can override the checked state', () => {
	class MockComponent extends React.Component {
		constructor(props) {
			super(props);
			this.state = { checked: false };
			this.changeChecked = this.changeChecked.bind(this);
		}

		changeChecked() {
			this.setState({ checked: true });
		}

		render() {
			return (
				<a onClick={ this.changeChecked }>
					<Checkbox label='foo' checked={ this.state.checked } />
				</a>
			);
		}
	}

	const test = mount(<MockComponent />);

	test.find('a').simulate('click');

	expect(test.find('.sidecat-checkbox').hasClass('checked')).toBe(true);
});