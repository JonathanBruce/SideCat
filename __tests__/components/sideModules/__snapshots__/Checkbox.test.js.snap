// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox changes the class when it is checked by default 1`] = `
<div
  className="sidecat-checkbox checked"
>
  <div
    className="checkbox"
    onClick={[Function]}
  >
    <div
      className="check"
    />
  </div>
  <label>
    foo
  </label>
</div>
`;
